connexion/decorators/uri_parsing.py

```diff
@@ -119,6 +120,15 @@ def resolve_params(self, params, _in):
             else:
                 resolved_param[k] = values[-1]
 
+            if not (is_nullable(param_defn) and is_null(resolved_param[k])):
+                try:
+                    # TODO: coerce types in a single place
```

Comment:
We now coerce the types in three locations
- Here, in the URI parser
- In the parameter validation
- In the form validation

I feel like we should be able to only do this in the URI parsing, but then we need some more work to still propagate the validation errors properly and make sure the URI parsing is used in all cases.